c7n/resources/ec2.py

```diff
@@ -634,6 +634,69 @@ def process_volume_set(self, resource):
                 Tags=tags)
 
 
+@actions.register('rename-tag')
+class RenameTag(BaseAction):
+
+    schema = type_schema(
+        'rename-tag',
+        source={'type': 'string'},
+        dest={'type': 'string'})
+
+    def process_ec2(self, client, instance):
+        """
+        Move source tag value to destination tag value
+
+        - Collect value from old tag
+        - Delete old tag
+        - Create new tag & assign stored value
+        """
+
+        source_tag = self.data.get('source')
+        tag_map = {t['Key']: t for t in instance.get('Tags', [])}
+        source = tag_map[source_tag]
+        destination_tag = self.data.get('dest')
+
+        client.delete_tags(
```

Comment:
of course to do that safely, you'll need to sanity check that the resource is under tag count limit.